src/clients/python/library/tritonclient/http/__init__.py

```diff
@@ -189,8 +192,11 @@ def __init__(self,
                  ssl_options=None,
                  ssl_context_factory=None,
                  insecure=False):
-        scheme = "https://" if ssl else "http://"
-        self._parsed_url = URL(scheme + url)
+        if not url.startswith("http://") and not url.startswith("https://"):
```

Comment:
Did you mean to add this? So putting the scheme in the URL overrides the 'ssl' setting? We don't document that override and we don't document that you can put the scheme in the url. Is the behavior the same as in C++? It seems like we should instead check to make sure that the URL does not have any *:// prefix and raise an error if it does.

Reply:
Good point. I have added a check to raise an error upon detecting the scheme in the url. 
I also found that we are missing the full-fledged SSL support in our C++ http client library. But when added would do something similar. I have added a ticket to support the certificate authentication in C++.